lib/child_process.js

```diff
@@ -50,11 +54,21 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   args = execArgv.concat([modulePath], args);
 
-  if (!Array.isArray(options.stdio)) {
+  if (typeof options.stdio === 'string') {
+    switch (options.stdio) {
+      case 'ignore':
+      case 'pipe':
+      case 'inherit':
+        options.stdio = stdioStringToArray(options.stdio);
+        break;
+      default:
+        throw new TypeError('Unknown stdio option');
```

Comment:
I think we should use the same [error string used by `_validateStdio()`](https://github.com/nodejs/node/blob/92eadd3037412293d593f7b9dd3bec99a29ad719/lib/internal/child_process.js#L773) which used by `spawn()`.